sourmash/commands.py

```diff
@@ -440,30 +443,23 @@ def compare(args):
     notify('')
 
     # build the distance matrix
-    D = numpy.zeros([len(siglist), len(siglist)])
     numpy.set_printoptions(precision=3, suppress=True)
 
     # do all-by-all calculation
-    labeltext = []
-    for i, E in enumerate(siglist):
-        for j, E2 in enumerate(siglist):
-            if i < j:
-                continue
-            similarity = E.similarity(E2, args.ignore_abundance)
-            D[i][j] = similarity
-            D[j][i] = similarity
 
-        labeltext.append(E.name())
+    labeltext = [sig.name() for sig in siglist]
```

Comment:
now, THAT'S a fun bug! Somehow this `sig` here replaces the `sig` module imported in the beginning of the file, and so the loop works for the first file but fails for the second!

If you change the name to something else, then everything works:
```suggestion
    labeltext = [item.name() for item in siglist]
```

And that's very confusing behavior! I'll try to find the source, but I'm pretty sure Python 3 change the lexical scope for this and then it's not a bug anymore (and the reason why it works in 3.x, but not in 2.7)